src/cow-react/modules/limitOrders/utils/getOrderExecutedAmounts.ts

```diff
@@ -0,0 +1,28 @@
+import { Order } from 'state/orders/actions'
+import JSBI from 'jsbi'
+
+/**
+ * Syntactic sugar to get the order's executed amounts as a BigNumber (in atoms)
+ * Mostly because `executedSellAmount` is derived from 2 fields (at time or writing)
+ *
+ * @param order The order
+ */
+export function getOrderExecutedAmounts(order: Order): {
+  executedBuyAmount: JSBI
+  executedSellAmount: JSBI
+} {
+  if (!order.apiAdditionalInfo) {
+    return {
+      executedBuyAmount: JSBI.BigInt(0),
+      executedSellAmount: JSBI.BigInt(0),
+    }
+  }
+
+  return {
+    executedBuyAmount: JSBI.BigInt(order.apiAdditionalInfo.executedBuyAmount),
```

Comment:
too much repetition of `apiAdditionalInfo`, once you asserted the existence of this, can't you just destrucure it and get `executedBuyAmount`, `executedSellAmount` and `executedFeeAmount`

Reply:
Good point, I was in a hurry to modify code from Explorer and to just get this to work but this should be cleaner. Updated.